library/ecp.c

```diff
@@ -1911,7 +1984,7 @@ int mbedtls_ecp_check_privkey( const mbedtls_ecp_group *grp, const mbedtls_mpi *
 
     return( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
 }
-
+#if ( ! defined ( MBEDTLS_ECP_GEN_KEYPAIR_BASE_ALT ) && ! defined ( MBEDTLS_ECP_MUL_ALT ) )
```

Comment:
What's the reason to also guard this by `MBEDTLS_ECP_MUL_ALT`? I just tried to compile the library with a stub `mbedtls_ecp_mul`, setting `MBEDTLS_ECP_MUL_ALT` but not `MBEDTLS_ECP_GEN_KEYPAIR_BASE_ALT`, which failed because `mbedtls_ecp_gen_keypair_base` is not defined.